src/js/common/components/form-elements/GrowableTable.jsx

```diff
@@ -52,21 +52,29 @@ class GrowableTable extends React.Component {
     this.setState({ [blankRowData.key]: 'incomplete' });
   }
 
+  findIncomplete() {
+    return _.findKey(this.state, v => v === 'incomplete');
+  }
+
   handleAdd() {
+    // Save existing
+    this.handleSave();
+
     if (this.props.isValidSection(this.props.path, this.props.data)) {
       this.createNewElement();
     }
   }
 
   handleEdit(event) {
-    this.setState({ [event.target.dataset.key]: 'incomplete' });
+    this.setState({ [event.target.dataset.key]: 'edit' });
```

Comment:
I saw one instance where clicking the edit button did nothing, but haven't been able to reproduce it.